src/datasets/utils/metadata.py

```diff
@@ -126,14 +126,30 @@ class MetadataConfigs(Dict[str, dict]):
 
     __configs_field_name: ClassVar[str] = METADATA_CONFIGS_FIELD
 
+    @classmethod
+    def _raise_if_not_valid(self, metadata_config: dict):
+        if isinstance(metadata_config.get("data_files"), dict):
+            raise ValueError(
+                f"Expected data_files in YAML to be a string or a list, but got {metadata_config['data_files']}\nExamples:\n"
+                "    data_files: data.csv\n    data_files: data/*.png\n"
+                "    data_files:\n    - part0/*\n    - part1/*\n"
+                "    data_files:\n    - split: train\n      pattern: train/*\n    - split: test\n      pattern: test/*"
+            )
+
     @classmethod
     def from_metadata(cls, dataset_metadata: DatasetMetadata) -> "MetadataConfigs":
-        if cls.__configs_field_name in dataset_metadata:
-            metadata_configs = dataset_metadata.get(cls.__configs_field_name)
+        if dataset_metadata.get(cls.__configs_field_name):
```

Comment:
```suggestion
        if dataset_metadata.get(cls.__configs_field_name, None):
```
it might not exist in readme

Reply:
`.get()` already returns None by default if the key is missing

Reply:
sorry i'm dumb :D